File: 125-valid-palindrome/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l = 0
        r = len(s)-1
        while l < r :

            while l < r and not self.IsAlNum(s[l]):
                l += 1

            while r > l and not self.IsAlNum(s[r]):
                r -= 1

            if s[l].lower() != s[r].lower():
                return False
            else:
                l, r = l + 1, r - 1

        return True

    def IsAlNum(self , c):

        if (ord('A') <= ord(c) <=ord('Z')  or
           ord('a') <=  ord(c) <= ord('z') or
           ord('0') <= ord(c) <= ord('9')):
            return True
        else:
            return False

File: 125-valid-palindrome/125-valid-palindrome.cpp
class Solution {

public:

    
bool solve(char c){

	if ( (int(c) >= 48 && int(c) <= 57) ||
		 (int(c) >= 65 && int(c) <= 90) ||
		 (int(c) >= 97 && int(c) <= 122) )
	{
		return true;
	}
	else
		return false;
}

    
    bool isPalindrome(string s) {
        
        
	int l = 0 ;
	int r = (int)s.size()-1 ;

	while(l < r ){

		while (l < r && !solve(s[l])){
			l ++;
		}

		while(r > l && !solve(s[r])){
			r--;
		}

		if (tolower(s[l]) != tolower(s[r])){
			return false ;
		}
		else {
			l++;
			r--;
		}
	}

	return true;

        
    }
};